docs: update chain.rst to improve discussion of upstreams (#43918)
LLNL/spack: lib/spack/docs/chain.rst

You can point your Spack installation to another installation to use any
packages that are installed there. To register the other Spack instance,

> you can add it as an entry to ``path/to/spack/etc/spack/upstreams.yaml``,
> if ```upstreams.yaml``` is not there, crete it similar to:

Reviewer: This isn't the only location where `upstreams.yaml` can be located. Any of the configuration scopes at https://spack.readthedocs.io/en/latest/configuration.html#configuration-scopes can support this. How about this?
```suggestion
you can add it as an entry to ``upstreams.yaml`` at any of the
:ref:`configuration-scopes`:
```
This will look similar to the end of the introduction section at https://spack.readthedocs.io/en/latest/packages_yaml.html.

    spack-instance-2:
      install_tree: /path/to/another/spack/opt/spack

> ``install_tree`` must point to the ``.../spack/opt/spack`` directory inside of 
> the Spack base directory.

Reviewer: This is incorrect, since there is no `.../spack/opt/spack` directory inside of the Spack base directory. The original is clearer; not every install of spack lives in a directory `spack`. Many live in e.g. `v0.22.0` to allow multiple versions to co-exists.

The documentation can be improved, however, by making sure it clear that you can point to any `install_tree`, not just the default location in `opt/spack`.

How about this?
```suggestion
``install_tree`` must point to the ``opt/spack`` directory inside of the
 Spack base directory, or the location of the ``install_tree`` defined
 in :ref:`config.yaml <config-yaml>`.
```

Author: I do like it, it is clear. `/opt` is general enogh

Reviewer: If you agree, please feel free to commit these changes (and address the other comment). Then I can review/approve and set this to merge.
